Task3/js/app.js

```diff
@@ -0,0 +1,109 @@
+const container = document.querySelector('.zombies');
+const status = document.querySelector('.zombies-status');
+const numberZombie = document.querySelectorAll('.zombies-status span')[0];
+const allZombies = document.querySelectorAll('.zombies-status span')[1];
+let counter = 0;
+let zombie;
+
+allZombies.textContent = zombies.length;
+numberZombie.textContent = counter + 1;
+
+function Zombie() {
+    this.health = 0;
+    this.healthCurrent = 0;
+    this.healthPercent = 100;
+    this.hitDamage = HIT_DAMAGE;
+
+    this.getDamage = () => {
+        if (this.healthCurrent > 0) {
+            this.healthCurrent -= this.hitDamage;
+        }
+
+
+        this.calculateHealth();
+    };
+
+    this.calculateHealth = () => {
+        this.healthPercent = this.healthCurrent / this.health  * 100;
+
+        if (this.healthPercent < 0) {
+            this.healthPercent = 0;
+        }
+    }
+}
+
+function SmallZombie() {
+    Zombie.call(this);
+
+    this.health = zombies[ZOMBIE_TYPE.SMALL].health;
+    this.healthCurrent = this.health;
+}
+
+function MadZombie() {
+    Zombie.call(this);
+
+    this.health = zombies[ZOMBIE_TYPE.MAD].health;
+    this.healthCurrent = this.health;
+}
+
+function StrongZombie() {
+    Zombie.call(this);
+
+    this.health = zombies[ZOMBIE_TYPE.MAD].health;
+    this.healthCurrent = this.health;
+}
+
+function createZombie() {
+    if (zombies.length && zombies.length - 1 >= counter) {
+        const zombieElement = document.createElement('div');
+        const zombieHealth = document.createElement('div');
+        const zombieHealthCurrent = document.createElement('div');
+
+        zombieElement.classList.add('zombie');
+        zombieHealth.classList.add('zombie-health');
+        zombieHealthCurrent.classList.add('zombie-health-current');
+
+        if (zombies[counter].type === ZOMBIE_TYPE.SMALL) {
+            zombie = new SmallZombie();
+
+            zombieElement.classList.add('small');
+        } else if (zombies[counter].type === ZOMBIE_TYPE.MAD) {
+            zombie = new MadZombie();
+
+            zombieElement.classList.add('mad');
+        } else if (zombies[counter].type === ZOMBIE_TYPE.STRONG) {
+            zombie = new StrongZombie();
+
+            zombieElement.classList.add('strong');
+        }
+
+        zombieElement.addEventListener('click', () => {
```

Comment:
same here,
please move the whole logic (events, methods, ...) into class-function
app file is too overloaded